core/src/main/java/org/elasticsearch/search/fetch/matchedqueries/MatchedQueriesFetchSubPhase.java

```diff
@@ -43,7 +43,7 @@
 
     @Override
     public void hitsExecute(SearchContext context, InternalSearchHit[] hits) {
-        if (hits.length == 0) {
+        if (hits.length == 0 || context.parsedQuery() == null) {
```

Comment:
hmm is that true? i wonder if this is equivalent to a match all? I could be very wrong.